### PR TITLE
[SourceKit] Save clang module validation time

### PR DESCRIPTION
**Pull request by rintaro**
Pass `-fbuild-session-timestamp` and `-fmodules-validate-once-per-build-session` to ClangImporter so that the module validation happens only once for the SourceKit lifetime.

rdar://problem/59567281

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by akyrtzi** [comment]
> sourcekitd-test -req=sema <..> == -copy-file <..> == -req=sema <..>

The second `sema` invocation probably needs some extra '-D' argument to force creation of a new AST, make sure it's not just re-using the first AST. Generally this test line should be failing without your changes.

**Comment by akyrtzi** [comment]
Probably also good to do a similar test using `complete` request, to verify we avoid system header validation for code-completion.

**Comment by benlangmuir** [comment]
We should also make sure we have a test for modifying a user header, where we expect to see the changes immediately.

**Comment by rintaro** [comment]
@swift-ci Please test

File: test/SourceKit/CodeComplete/complete_build_session.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// CHECK_SYSTEM_2-DAG: key.description: "fooHelperFunc_mod(arg: Int32)"
+// CHECK_SYSTEM_2-DAG: key.description: "fooHelperSubFunc_mod(arg: Int32)"
+// CHECK_SYSTEM_2-NOT: fooFunc(
+// CHECK_SYSTEM_2-NOT: fooSubFunc(
+// CHECK_SYSTEM_2-NOT: fooHelperFunc(
+// CHECK_SYSTEM_2-NOT: fooHelperSubFunc(
+
+// -----------------------------------------------------------------------------
+// Test that modifications for frameworks in '-F' are immidiately propagated
+// while modifications for frameworks in '-F' is not.
```
Comment (akyrtzi) [inline test/SourceKit/CodeComplete/complete_build_session.swift]: Did you mean to say '-Fsystem' here? Same in the other test file.

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please test

**Review by akyrtzi — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please test Windows

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
Windows testing is running in https://ci-external.swift.org/job/swift-PR-windows/996/

**Comment by rintaro** [comment]
Windows failure is unrelated. Merging

**Pull request merged**